packages/core/src/actions/emit.ts

```diff
@@ -0,0 +1,152 @@
+import isDevelopment from '#is-development';
+import {
+  ActionArgs,
+  AnyActorScope,
+  AnyActor,
+  AnyMachineSnapshot,
+  DelayExpr,
+  EventObject,
+  MachineContext,
+  NoInfer,
+  RaiseActionOptions,
+  SendExpr,
+  ParameterizedObject,
+  AnyEventObject
+} from '../types.ts';
+
+function resolveEmit(
+  _: AnyActorScope,
+  snapshot: AnyMachineSnapshot,
+  args: ActionArgs<any, any, any>,
+  actionParams: ParameterizedObject['params'] | undefined,
+  {
+    event: eventOrExpr,
+    id,
+    delay
+  }: {
+    event:
+      | EventObject
+      | SendExpr<
+          MachineContext,
+          EventObject,
+          ParameterizedObject['params'] | undefined,
+          EventObject,
+          EventObject
+        >;
+    id: string | undefined;
+    delay:
```

Comment:
If u plan to support `delay` here then you will have to account for that in the whacky persistence that we have today. Since this is additive I'd start without it to simplify things. And at that point you won't need `id` either